salt/state.py

```diff
@@ -4254,7 +4254,7 @@ def load_dynamic(self, matches):
         """
         if not self.opts["autoload_dynamic_modules"]:
             return
-        syncd = self.state.functions["saltutil.sync_all"](list(matches), refresh=False)
+        syncd = self.state.functions["saltutil.sync_all"](list(matches), refresh=True)
```

Comment:
I'm concerned about changes this to True.

This was changed to False here: https://github.com/saltstack/salt/pull/12195 to fix this issue https://github.com/saltstack/salt/issues/12173

Are we going to be re-introducing this behavior with this change?

Reply:
The original code had false for local and true for remote, and then @terminalmage made the change and it was false for both.  So originally this was for a speed up when local, but either way it doesn't make sense to me, given I just sync'd and updated and possibly picked up new grains (which is the case here with _grains), would you not want those refreshed ?  That is why when I originally submitted this PR, I pointedly asked the team to review this statement and to consider whether the reason's are still valid 10 years later.